### PR TITLE
add prefresh capabilities

### PR DESCRIPTION
**Pull request by JoviDeCroock**
This fixes compatability with `next 9.4` by removing `react-refresh` and using `prefresh`

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -28,13 +30,20 @@ module.exports = {
...
       config.entry = () => entry().then(entries => {
         entries['main.js'] = ['preact/debug'].concat(entries['main.js'] || []);
         return entries;
       });
+
+      // Inject Prefresh
+      const reactRefresh = config.plugins.find(s => s.constructor.name === 'ReactFreshWebpackPlugin');
+      if (reactRefresh) {
+        config.plugins.splice(config.plugins.indexOf(reactRefresh), 1);
+      }
```
Comment (JoviDeCroock) [inline next.config.js]: Ideally we'd have an else branch here to add this when we are at pre 9.4 but couldn't figure out how

File: pages/about.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import React from 'react'
 
-export default () => (
+const About = () => (
   <main>
     <h1>About us</h1>
   </main>
 );
 
+export default About;
```
Comment (srdjan) [inline pages/about.js:9]: btw, is this somehow related to 'prefersh', or just cosmetic?

Comment (JoviDeCroock) [reply pages/about.js:9]: This is related to both react-refresh and prefresh because the function needs a name to be able to determine whether or not it's a component

File: pages/index.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
 import Link from 'next/link'
 
-export default () => (
+const Home = () => (
```
Comment (chrsep) [inline pages/index.js]: Hi, I tried running the demo, it seems that adding a hook stopped prefresh from working. I tested it by adding a `useState(0)` here. 

I was tinkering with prefresh on my own nextjs project and ran into a similar problem and I can't seem to figure out why 😅.  Does it have something to do with `react-refresh/babel`? 

I love that a nextjs plugin is coming out so fast btw, thx for working on it.

Comment (JoviDeCroock) [reply pages/index.js]: It does seem to work correctly for me, what version of next are you using? If it's pre 9.4 this won't work unless you add the babel plugin yourself

Comment (chrsep) [reply pages/index.js]: Ah, apparently it's not caused by `@prefresh/nextjs`. It seems to be caused by preact-devtools on my end. I just tried it with preact-devtools plugin turned off and it runs perfectly. If I commented out the devtools injection code on `next.config.js`, it would also work. I'm using 9.4.

Comment (JoviDeCroock) [reply pages/index.js]: Hmm odd that, going to try and figure out what's going wrong

Comment (developit) [reply pages/index.js]: I wonder if a hook is being injected but the original isn't being called?

File: package.json [hunk 4]
```diff
@@ -6,9 +6,13 @@
     "build": "next build && next export -o build",
     "start": "NODE_ENV=production next start"
   },
+  "devDependencies": {
+    "@prefresh/next": "^0.2.0",
```
Comment (aralroca) [inline package.json]: @JoviDeCroock I tried with `@prefresh/next@0.3.0` and I got an error:

image

However, downgrading to `0.2.0` is working well. Do you know why that would happen?

Comment (JoviDeCroock) [reply package.json]: The import maps must be wrong for your node version, which one are you using?

Comment (aralroca) [reply package.json]: I'm using node 14.3.0

Comment (developit) [reply package.json]: `@prefresh/core` is being imported as commonjs (because @prefresh/next [is](https://github.com/developit/nextjs-preact-demo/pull/16/files#diff-5d0c276360a637d1b787a57760665fbeR1)), but the main export (`"."`) is only defined as supporting ESM: https://github.com/JoviDeCroock/prefresh/blob/51c583b9f2e830ac825e024428213821be1f82b2/packages/core/package.json#L7

Adding a `"default"` export would fix this.

Comment (aralroca) [reply package.json]: I can confirm that this was the problem! Thanks @developit @JoviDeCroock

Comment (JoviDeCroock) [reply package.json]: @aralroca

Did you add `default` to the export map?

Comment (aralroca) [reply package.json]: Yes, I touched directly in the `node_modules/@prefresh/core/package.json` on my project, and it works pretty well:

```diff
"exports": {
    ".": {
      "import": "./src/index.js",
+      "default": "./src/index.js"
    },
    "./package.json": "./package.json",
    "./": "./"
  },
```

I forked now the prefesh repo to do the PR, but you got faster 😊

**Comment by JoviDeCroock** [comment]
Currently there's a known issue where `devtools` and `prefresh` don't cooperate well, the pre-refresh hookState is kept around.

**Comment by JoviDeCroock** [comment]
The issue with `preact-devtools` is fixed with the 1.0.0 extension releasing! 🎉

File: pages/index.js [hunk 5]
```diff
@@ -1,12 +1,18 @@
 import React from 'react'
 import Link from 'next/link'
 
-export default () => (
-  <main>
-    <h1>Hello from Preact</h1>
+const Home = () => {
```
Comment (JoviDeCroock) [inline pages/index.js:4]: Due to `export default () => ` having no name this is needed to make `prefresh` recognise this as a component as said in our [best practices section](https://github.com/JoviDeCroock/prefresh#best-practices)

**Review by developit — APPROVED** [review]

File: pages/index.js (hunk 5 above)
Comment (developit) [reply pages/index.js:4]: Seems fine. I'd love to have this be able to infer components from the existence of JSX within them, but this is a very reasonable compromise.
